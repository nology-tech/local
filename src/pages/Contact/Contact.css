@import url("https://fonts.googleapis.com/css2?family=Inter&family=Lato&family=Roboto&display=swap");
* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

/*                          Colours                         */
/*                          Font stuff              */
* {
  font-family: "Trebuchet MS", "Lucida Sans Unicode", "Lucida Grande", "Lucida Sans", Arial, sans-serif;
  color: #000000;
}

h1,
h2,
h3,
h4,
h5,
h6,
p {
  margin: 0.5rem 0;
  display: inline-block;
  width: 100%;
}

button {
  font-size: 1rem;
}

h1 {
  font-size: 2.1rem;
}

h2 {
  font-size: 1.65rem;
}

h3 {
  font-size: 1.25rem;
}

@media screen and (min-width: 768px) {
  :root {
    font-size: 110%;
  }
}
@media screen and (min-width: 1024px) {
  :root {
    font-size: 120%;
  }
}
.contact {
  height: 100%;
  background: rgb(236, 148, 114);
  background: linear-gradient(152deg, rgb(236, 148, 114) 0%, rgb(224, 95, 30) 100%);
  padding: 1rem;
}
.contact .contact__form {
  display: flex;
  flex-flow: column;
  justify-content: space-evenly;
  align-items: center;
  margin-bottom: 1rem;
}
.contact .contact__form > *:last-child {
  margin-top: 1rem;
}/*# sourceMappingURL=Contact.css.map */