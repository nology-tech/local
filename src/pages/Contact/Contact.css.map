{"version":3,"sources":["../../styles/_typography.scss","../../styles/_reset.scss","Contact.css","../../styles/_variables.scss","Contact.scss","../../styles/_mixins.scss"],"names":[],"mappings":"AACQ,oGAAA;ACDR;EACE,UAAA;EACA,SAAA;EACA,sBAAA;ACEF;;ACEA,6DAAA;AAuBA,qDAAA;AHxBA;EACE,qGAAA;EAEA,cGCW;ADEb;;AFAA;;;;;;;EAQE,gBAAA;EACA,qBAAA;EACA,WAAA;AEEF;;AFCA;EACE,eAAA;AEEF;;AFCA;EACE,iBAAA;AEEF;;AFCA;EACE,kBAAA;AEEF;;AFCA;EACE,kBAAA;AEEF;;AFCA;EAEE;IACE,eAAA;EECF;AACF;AFEA;EAEE;IACE,eAAA;EEDF;AACF;AE9CA;EACE,YAAA;ECIA,8BFIW;EEHX,iFFIoB;ECPpB,aAAA;AFiDF;AEhDE;EACE,aAAA;EACA,iBAAA;EACA,6BAAA;EACA,mBAAA;EACA,mBAAA;AFkDJ;AEjDI;EACE,gBAAA;AFmDN","file":"Contact.css"}