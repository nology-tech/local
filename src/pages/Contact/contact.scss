@use "../../../src/styles/reset" as *;
@use "../../../src/styles/variables" as *;
@use "../../styles/typography" as *;
@use "../../styles/mixins" as *;

.contact-container {
  height: 100vh;
  background-color: orangered;
  width: 100vw;
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  .contact-form {
    width: 100%;
    padding: 0 1rem 0.5rem 1rem;
    display: flex;
    flex-flow: column;
    justify-content: space-evenly;
    align-items: center;
    &__header,
    &__paragraph {
      padding-left: 1rem;
    }
  }
}
