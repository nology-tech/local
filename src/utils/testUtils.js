import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
