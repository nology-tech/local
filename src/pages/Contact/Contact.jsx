import Layout from "../../components/Layout/Layout";
import Button from "../../components/button/Button";
import "./contact.scss";
import TextField from "../../components/TextField/TextField";

const Contact = () => {
  return (
    <Layout>
      <main className="contact">
        <h1>Contact</h1>
        <p>We would love to hear from you and take your feedback on board.</p>
        <form className="contact__form">
          <TextField
            uniqueInputFieldName="contactName"
            inputType="text"
            labelHeader="Name"
          />
          <TextField
            uniqueInputFieldName="contactEmail"
            inputType="email"
            labelHeader="Email"
          />
          <TextField
            uniqueInputFieldName="contactMessage"
            inputType="text"
            labelHeader="Message"
            modifier="large"
          />
          <Button buttonText={"SEND"} buttonName="contact" />
        </form>
      </main>
    </Layout>
  );
};

export default Contact;
