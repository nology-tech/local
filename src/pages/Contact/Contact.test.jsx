import Contact from "./Contact";
import { customRender } from "../../utils/testUtils";

it("should render the layout component", () => {
  const { container } = customRender(
    <Layout>
      <p>Hello there</p>
    </Layout>
  );

  expect(container).toMatchSnapshot();
});
