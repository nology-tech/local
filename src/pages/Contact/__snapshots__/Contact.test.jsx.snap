// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render contact page 1`] = `
<div>
  <div
    class="layout"
  >
    <div
      class="layout__header"
    >
      <div
        class="header"
      >
        <img
          alt=""
          src="BackButton.svg"
        />
        <div
          class="header__container"
        >
          <div
            class="header__logo-title"
          >
            <img
              alt=""
              src="AppLogo.svg"
            />
            <h1
              class="header__title-local"
            >
              LOCAL
            </h1>
          </div>
        </div>
      </div>
      <img
        alt="Menu icon"
        src="menu.svg"
      />
    </div>
    <main
      class="contact"
    >
      <h1>
        Contact
      </h1>
      <p>
        We would love to hear from you and take your feedback on board.
      </p>
      <form
        class="contact__form"
      >
        <div
          class="text-field"
        >
          <label
            class="text-field__label"
          >
            Name
          </label>
          <input
            class="text-field__input"
            type="text"
          />
        </div>
        <div
          class="text-field"
        >
          <label
            class="text-field__label"
          >
            Email
          </label>
          <input
            class="text-field__input"
            type="email"
          />
        </div>
        <div
          class="text-field"
        >
          <label
            class="text-field__label"
          >
            Message
          </label>
          <input
            class="text-field__input text-field__input--large"
            type="text"
          />
        </div>
        <button
          class="button button--contact"
        >
          SEND
        </button>
      </form>
    </main>
  </div>
</div>
`;
